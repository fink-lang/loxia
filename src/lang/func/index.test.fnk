{fink2js} = import '../../testing'
{describe, it, eq, snapshot} = import '../../testing/jest'


describe:: 'func', fn:
  it:: 'compiles', fn:
    fink2js(`
      fun1 = fn a, b=12, ...d: a + b + c

      fun2 = fn b:
        b * 2

      fun3 = fn: 'shrub'
      fun4 = fn:
        'shrub'

      fun5 = fn c: fn d, e:
        match [d, e]:
          [1, 2]: c
          [2, 1]: 1/c

      fun6 = fn c, d:
        x = 123
        match ni:
          c + x: d
          c + 1: d + 1

      fun7 = fn a, b=12, c:
        shrub(...a)
        foo()
        bar()
    `) eq snapshot



