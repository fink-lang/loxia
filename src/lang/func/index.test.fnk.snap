// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`func compiles defaults 1`] = `
"export const fun = (a, b = 12, c) => {
  shrub(...a);
  foo();
  return bar();
};"
`;

exports[`func compiles empty arg 1`] = `"export const fun = (a, ˆ_1, ˆ_2, b) => b;"`;

exports[`func compiles func of func 1`] = `
"export const fun5 = (c) => (d, e) => {
  const ˆvalue_1 = [d, e];

  if (ˆvalue_1 !== undefined && ˆvalue_1 !== null && typeof ˆvalue_1[Symbol.iterator] === \\"function\\") {
    const [ˆa_3, ˆa_4] = ˆvalue_1;
    if (ˆa_3 === 1) if (ˆa_4 === 2) {
      return c;
    }
  }

  if (ˆvalue_1 !== undefined && ˆvalue_1 !== null && typeof ˆvalue_1[Symbol.iterator] === \\"function\\") {
    const [ˆa_5, ˆa_6] = ˆvalue_1;
    if (ˆa_5 === 2) if (ˆa_6 === 1) {
      return 1 / c;
    }
  }
};"
`;

exports[`func compiles other 1`] = `
"export const fun = (c, d) => {
  const x = 123;
  {
    const ˆvalue_1 = ni;

    if (ˆvalue_1 === c + x) {
      return d;
    }

    if (ˆvalue_1 === c + 1) {
      return d + 1;
    }
  }
};"
`;

exports[`func compiles single line 1`] = `"export const fun = () => \`shrub\`;"`;

exports[`func compiles spread 1`] = `"export const fun = (a, b = 12, ...d) => a + b + c;"`;
